File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    dateList = []
    dateValue = 0
    for i in range(0, len(progresses), 1):
        dateValue = (100-progresses[i])/speeds[i]
        dateList.append(math.ceil(dateValue))
    
    cnt = 1
    temp = dateList[0]
    for i in range(1, len(dateList), 1):
        if temp >= dateList[i]:
            cnt = cnt + 1
        else:
            answer.append(cnt)
            temp = dateList[i]
            cnt = 1            
    answer.append(cnt)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer = {};
        int[] days = new int[progresses.length];
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < progresses.length; i++) {
            days[i] = ((int) Math.ceil((double)(100 - progresses[i]) / speeds[i]));
        }
        
        ArrayList<Integer> answerList = new ArrayList<>();
        int idx = 0;
        while(idx < progresses.length){
            int count = 1;
            int curDay = days[idx];

            while(idx + count < progresses.length && days[idx + count] <= curDay) {
                count++;
            }
            answerList.add(count);
            idx += count;
        }
        stack.addAll(answerList);
        
        answer = new int[stack.size()];
        
        for (int i = stack.size() - 1; i >= 0; i--){
            answer[i] = stack.pop();
        }
        return answer;
    }
}